test/e2e/framework/statefulset_utils.go

```diff
@@ -193,11 +196,50 @@ func getStatefulSetPodNameAtIndex(index int, ss *apps.StatefulSet) string {
 	return fmt.Sprintf("%v-%v", ss.Name, index)
 }
 
+// newStatefulSetScale returns a new scale subresource for a statefulset.
+func newStatefulSetScale(ss *apps.StatefulSet, count int32) *extensions.Scale {
+	return &extensions.Scale{
+		// TODO: Create a variant of ObjectMeta type that only contains the fields below.
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ss.Name,
+			Namespace: ss.Namespace,
+		},
+		Spec: extensions.ScaleSpec{
+			Replicas: count,
+		},
+		Status: extensions.ScaleStatus{
+			Replicas: ss.Status.Replicas,
+			Selector: ss.Spec.Selector,
+		},
+	}
+}
+
 // Scale scales ss to count replicas.
 func (s *StatefulSetTester) Scale(ss *apps.StatefulSet, count int32) error {
 	name := ss.Name
 	ns := ss.Namespace
-	s.update(ns, name, func(ss *apps.StatefulSet) { *(ss.Spec.Replicas) = count })
+
+	// create a new scale subresource
+	scale := newStatefulSetScale(ss, count)
+
+	// TODO: add scale endpoint to StatefulSet interface
```

Comment:
@janetkuo @enisoc 
I want to have something like this: `err := s.c.AppsV1beta2().StatefulSets(ns).Scale(scale)`. It requires:
(1) implementation of `Scale` endpoint setter
(2) enable `Scale` type at `pkg/apis/apps/types.go` and `staging/src/k8s.io/api/apps/v1beta2/types.go` (done by Janet) by putting `// +genclient=true` annotation

Due to addition of annotation, i still need to run `hack/update-codegen.sh` to regenerate clientset, which should not be required. I think there is a gap between my understanding on adding an endpoint.

I had a try with PATCH. I am not sure whether this approach hits something that you guys want.